### PR TITLE
[1.9] Fuel handler improvement - allow to make fuel value smaller or 0

### PR DESCRIPTION
**Pull request by liach**
1.9 version of #2218 and #2186.
Created because Lex and cpw did not rewrite fuel handling code.

### CODE REVIEW

**Comment by ShetiPhian** [comment]
While I doubt this will be pulled, as cpw didn't want it in future versions, I'm going to bring this up again just in case he changes his mind.

The issues brought up in the original discussion still haven't been addressed.

cpw pointed out getFuelValue should iterate over the entire list because "Some mods use this for things other than simply returning a value."

Also the original system returns the highest burn value, while this uses the first one found.
Suggested fixed code: https://github.com/MinecraftForge/MinecraftForge/pull/1847#issuecomment-156746149

Don't get me wrong, other then the unnecessary breaking of existing functionally, I support the improvements/enhancements your pulls provide.

**Comment by liach** [comment]
> Some mods use this for things other than simply returning a value.

An example please.

File: src/main/java/net/minecraftforge/fml/common/registry/GameRegistry.java [hunk 1]
```diff
@@ -353,12 +355,16 @@ public static void registerFuelHandler(IFuelHandler handler)
...
         {
-            fuelValue = Math.max(fuelValue, handler.getBurnTime(itemStack));
+            int t = li.previous().getBurnTime(itemStack);
+            if (t > 0)
+                return t;
+            if (t < 0)
+                return 0;
         }
-        return fuelValue;
+        return TileEntityFurnace.getItemBurnTimeOld(itemStack);
```
Comment (kashike) [inline src/main/java/net/minecraftforge/fml/common/registry/GameRegistry.java]: Why is this using a furnace method?

Comment (liach) [inline src/main/java/net/minecraftforge/fml/common/registry/GameRegistry.java]: https://github.com/MinecraftForge/MinecraftForge/pull/2691/files#diff-4091e3de5a552e14f36d3897b35ec07bR66

**Pull request closed without merging**